src/application-base/private/useTestOverrides.js

```diff
@@ -0,0 +1,33 @@
+import { useState, useEffect } from 'react';
+
+const APPLICATION_BASE_OVERRIDE_EVENT = 'applicationBase.testOverride';
+
+/**
+ * Custom React hook that overrides locale via events. Leveraged by full stack testing to control deployed applications.
+ * @returns {Object} Contains the locale to override with.
+ */
+const useTestOverrides = () => {
+  const [localeOverride, setLocaleOverride] = useState();
+
+  const handleTestOverrides = (event) => {
```

Comment:
I'd move this function definition into the useEffect block. There's no need to recreate with every update.

Reply:
Could you also define it outside the useTestOverrides to not recreate it each update?

Reply:
oops good catch, moved into useEffect https://github.com/cerner/terra-application/pull/24/commits/1be4958a1fca0926deb7379a4ea5aab5848bf38f